parser/parser.go

```diff
@@ -107,6 +109,9 @@ type Parser struct {
 
 	// Attributes are attached to block level elements.
 	attr *ast.Attribute
+
+	// Current working directory for relative includes.
+	cwd string
```

Comment:
I can imagine some use cases where users of the package would want to provide more than one directory for resolving includes.

Another possibility is to have an option for fully virtualizing this. For example what if the user wanted to resolve the includes by extracting a file from a .zip file?

This could be achieved with a single, optional callback function like `func readInclude(path string) ([]byte, error)`.

We could provide a default implementation for file system that looks up in one or more directories, as an example.

This would also remove the need for logging if logging is only needed to notify about missing includes because the caller could implement `readInclude` callback and do application-specific logging.

Reply:
This *is* a good idea, see https://github.com/gomarkdown/markdown/pull/34
had to discard the returned error though and make it the IncludeHook's problem as if we have an error we still need to do something with it.